### DISCUSSION TITLE
Importing 799 Simple Orienteering Course Data

### DISCUSSION THREAD

**Discussion by landnavapp** [OpenOrienteering/mapper#2082] [Questions & Answers]
Greetings!

I am developing a new app which will automatically creates a randomized orienteering course, corresponding to some variables I've set (i.e., total distance, leg ranges etc).  The app successfully gives me a list of control coordinates in KML format, and I can then upload that KML file into OpenOrienteeringMapper, using the "assign new symbols" to import the data as a 799 Simple Orienteering Course.  So, the capabilities to import it into OpenOrienteeringMapper are there.

However, when I do this, I still need to manually change each point from "non-dash" point to "dash" point, so each bend in the course does not simply appear as a bend in the line, but as a control circle.  Manually changing each point into a circle means I have to zoom in on the map, and I don't want to have to look at the map before I print it or run it.  

Is there anything I can add to my KML file so that it automatically loads this data as 799 Simple Orienteering Course, and also so that it correctly imports each of these control points as "dash" points i.e. circles, rather than regular points?

Sample KML course input:   
`<?xml version="1.0" encoding="UTF-8"?>
`
`<kml xmlns="http://www.opengis.net/kml/2.2">
`
` <Document>
`
`  <Placemark>
`
`   <LineString>
`
`    <coordinates>
`
`	  -76.894417,38.993841,0
`
`	  -76.897368,38.990121,0
`
`	  -76.897995,38.988225,0
`
`	  -76.893808,38.988005,0
`
`	  -76.892706,38.990232,0
`
`	  -76.894360,38.992676,0
`
`	  -76.892885,38.993103,0
`
`	  -76.891097,38.993519,0
`
`	  -76.894299,38.986696,0
`
`	  -76.897303,38.984962,0
`
`	  -76.894417,38.993841,0
`
`	</coordinates>
`
`   </LineString>
`
`  </Placemark>
`
` </Document>
`
`</kml>
`

I examined the contents of an OpenStreetMap OSM file to see what it was that OpenOrienteeringMapper imported.  From what I can tell, it interprets the "TAG" fields and assigns them to corresponding values in the Symbol set.  However, I didn't see any way to assign a "dash" value to the coordinates.  Here are a few TAG fields I saw in a OSM file, for reference:

`  <tag k="highway" v="path"/>
`
`  <tag k="incline" v="up"/>
 `
` <tag k="mtb" v="yes"/>
 `
` <tag k="mtb:scale" v="1"/>
 `
` <tag k="mtb:scale:uphill" v="1"/>
 `
` <tag k="mtb:type" v="track"/>
 `
` <tag k="surface" v="dirt"/>
`  

 and for another trail:

`   <tag k="access" v="permit"/>
`
`  <tag k="bicycle" v="permit"/>
`
` <tag k="highway" v="track"/>
`

**Comment by dg0yt** [comment]
Thanks for your questions.

> Is there anything I can add to my KML file so that it automatically loads this data as 799 Simple Orienteering Course, and also so that it correctly imports each of these control points as "dash" points i.e. circles, rather than regular points?

I don't think we have this at the moment, unfortunately.

> I examined the contents of an OpenStreetMap OSM file to see what it was that OpenOrienteeringMapper imported. From what I can tell, it interprets the "TAG" fields and assigns them to corresponding values in the Symbol set. However, I didn't see any way to assign a "dash" value to the coordinates.

Mapper has a powerful language for assigning symbols based on OSM tags. However, this doesn't extend to turning individual points of a line object into "dash points" (to show the dash point symbol). It would be useful also for power lines.
